llvm-general/src/LLVM/General/Internal/Attribute.hs

```diff
@@ -50,15 +51,27 @@ instance Monad m => EncodeM m A.PA.ParameterAttribute (Ptr FFI.ParameterAttrBuil
       A.PA.InAlloca -> FFI.parameterAttributeKindInAlloca
       A.PA.NonNull -> FFI.parameterAttributeKindNonNull
       A.PA.Returned -> FFI.parameterAttributeKindReturned
+      A.PA.SwiftSelf -> FFI.parameterAttributeKindSwiftSelf
+      A.PA.SwiftError -> FFI.parameterAttributeKindSwiftError
+      A.PA.WriteOnly -> FFI.parameterAttributeKindWriteOnly
+      A.PA.Alignment _ -> error $ "llvm-general internal error: cases inconsistent in ParameterAttribute encoding for " ++ show a
+      A.PA.Dereferenceable _ -> error $ "llvm-general internal error: cases inconsistent in ParameterAttribute encoding for " ++ show a
+      A.PA.DereferenceableOrNull _ -> error $ "llvm-general internal error: cases inconsistent in ParameterAttribute encoding for " ++ show a
 
 instance Monad m => EncodeM m A.FA.FunctionAttribute (Ptr FFI.FunctionAttrBuilder -> EncodeAST ()) where
   encodeM (A.FA.StringAttribute kind value) = return $ \b -> do
     (kindP, kindLen) <- encodeM kind
     (valueP, valueLen) <- encodeM value
     liftIO $ FFI.attrBuilderAddStringAttribute b kindP kindLen valueP valueLen
-  encodeM a = return $ \b -> liftIO $ case a of
-    A.FA.StackAlignment v -> FFI.attrBuilderAddStackAlignment b v
-    _ -> FFI.attrBuilderAddFunctionAttributeKind b $ case a of
+  encodeM a = return $ \b -> case a of
+    A.FA.StackAlignment v -> liftIO $ FFI.attrBuilderAddStackAlignment b v
+    A.FA.AllocSize x y -> do x' <- encodeM x
+                             y' <- encodeM y
+                             liftIO $ FFI.attrBuilderAddAllocSize b x' y'
```

Comment:
formatting:
multi-line do-blocks wrapped after the "do", indented with as few units of 2 spaces as still works.